apps/labs/posts/implement-lapack-routines-in-stdlib.md

```diff
@@ -0,0 +1,210 @@
+---
+title: "Adding support for LAPACK routines in stdlib"
+authors: [pranav-goswami]
+published: September 30, 2024
+description: "Implementing LAPACK routines for numerical computation in web applications"
+category: [stdlib, mathematics, linear-algebra, numerical-computing]
+featuredImage:
+  src: /posts/hello-world-post/featured.png
+  alt: "WIP"
+hero:
+  imageSrc: /posts/hello-world-post/hero.jpeg
+  imageAlt: "WIP"
+---
+
+# Adding support for LAPACK routines in stdlib
+
+Hello, I am Pranav Goswami, a Computer Science graduate from IIT Jodhpur. I have been contributing to various open source organisations primarily LFortran and stdlib.
+
+### About project
+
+During the course of internship my goal was to add support for as many LAPACK routines to stdlib as possible.
+
+<!-- ![alt text](image02.jpg) -->
+
+<img src="/posts/implement-lapack-routines-in-stdlib/image02.jpg" alt="alt text" style="position:relative;left:10%;width:80%;height:400px;">
+
+<br>
+
+<!-- ![alt text](image.png) -->
+
+<!-- < insert a graphic or gif where a stickman picks a package and drops to another place > -->
+
+Now, it might seem what's tricky in that, just take existing Fortran implementation, translate it to javascript ( shh, via chatGPT? ), follow stdlib conventions, do benchmarking, add tests, documentation, etc and you are done. Sounds simple, but there is a catch or I say there are multiple catches, please read through the blog to get a detailed walkthrough.
+
+### Motivation
+
+Fortran has long been a foundational programming language for scientific computing, while JavaScript dominates the web ecosystem. I’ve observed various organizations attempting to compile Fortran codebases into WebAssembly (Wasm) for browser execution. This is where I believe stdlib is simplifying the process by offering APIs that enable execution directly via Node.js in a web environment. Leveraging the JavaScript standard library for direct execution on the web offers significant performance advantages, primarily by eliminating the need for implicit data transfers between WebAssembly (Wasm) and JavaScript. This approach also reduces the number of floating-point operations (FLOPs), and by keeping computations within JavaScript, higher performance is ensured. Additionally, these routines are beneficial for various IoT applications that lack Wasm support, making JavaScript an optimal choice in such contexts. This approach intrigues me, which is why I chose to explore it further.
+
+<!-- More reasons of why this projecT:
+1. Innvoation aspect: currently BLIS only supports BLAS
+2. There can be performance advantages incomparision to copying data to Wasm and back to JS. If always can be in js, always better performance, as it is more native.
```

Comment:
This comment isn't always true. Depends on the number of FLOPS. E.g., for `matmul`, copying to Wasm memory can still be advantageous, as O(N^3), while copying to and from is O(N^2).

Reply:
I removed it, we don't it here anyways.